src/main/java/com/google/summit/translation/Translate.kt

```diff
@@ -366,23 +367,21 @@ class Translate(val file: String, private val tokens: TokenStream) : ApexParserB
   override fun visitTypeList(ctx: ApexParser.TypeListContext): List<TypeRef> =
     ctx.typeRef().map { visitTypeRef(it) }
 
-  /** Translates the 'classBody' grammar rule and returns an AST [Declaration] list. */
-  override fun visitClassBody(ctx: ApexParser.ClassBodyContext): List<Declaration> =
-    ctx.classBodyDeclaration().flatMap { visitClassBodyDeclaration(it) }
+  /** Translates the 'classBody' grammar rule and returns an AST [Node] list. */
+  override fun visitClassBody(ctx: ApexParser.ClassBodyContext): List<Node> =
+    ctx.classBodyDeclaration().mapNotNull { visitClassBodyDeclaration(it) }
 
-  /**
-   * Translates the 'classBodyDeclaration' grammar rule and returns an optional AST [Declaration].
-   */
-  override fun visitClassBodyDeclaration(
-    ctx: ApexParser.ClassBodyDeclarationContext
-  ): List<Declaration> {
+  /** Translates the 'classBodyDeclaration' grammar rule and optionally returns an AST [Node]. */
+  override fun visitClassBodyDeclaration(ctx: ApexParser.ClassBodyDeclarationContext): Node? {
     matchExactlyOne(ruleBeingChecked = ctx, ctx.memberDeclaration(), ctx.block(), ctx.SEMI())
 
     return when {
       ctx.memberDeclaration() != null -> {
-        val members = visitMemberDeclaration(ctx.memberDeclaration())
-        members.forEach { member -> member.modifiers = ctx.modifier().map { visitModifier(it) } }
-        members
+        val member = visitMemberDeclaration(ctx.memberDeclaration())
+        if (member is Modifiable) {
```

Comment:
This will always be `Modifiable`, right? thoughts on an exception if it unexpectedly isn't?

Reply:
That sounds good to me. Intersection types would have definitely been useful (i.e. `ClassDeclaration` would have a `List<Node & Modifiable>`) but it seems like using `Node` and asserting that it's `Modifiable` when needed is the best remaining option.